pandas/core/indexes/base.py

```diff
@@ -986,18 +980,34 @@ def astype(self, dtype, copy: bool = True):
                 new_values = astype_array(values, dtype=dtype, copy=copy)
 
         # pass copy=False because any copying will be done in the astype above
-        if self._is_backward_compat_public_numeric_index:
-            # this block is needed so e.g. NumericIndex[int8].astype("int32") returns
-            # NumericIndex[int32] and not Int64Index with dtype int64.
+        if not self._is_backward_compat_public_numeric_index and not isinstance(
```

Comment:
so this chunk of code is going to be removed in a follow-up right?

Reply:
Yes, It will be removed in the followup #50052.